src/SSVOpenHexagon/Core/HexagonGame.cpp

```diff
@@ -254,7 +257,7 @@ void HexagonGame::incrementDifficulty()
     levelStatus.rotationSpeed *= -1.f;
 
     const auto& rotationSpeedMax(levelStatus.rotationSpeedMax);
-    if(status.fastSpin < 0 && abs(levelStatus.rotationSpeed) > rotationSpeedMax)
+    if(abs(levelStatus.rotationSpeed) > rotationSpeedMax)
```

Comment:
This is the one that I do not understand. This change prevents a "fast spin" from (temporarily) going over the rotation speed max. However, `status.fastSpin` should eventually go back down to `0` due to `status.fastSpin -= mFT;` in `HexagonGame::updateRotation`, so the rotation cap should be applied after the "fast spin" is over.

Maybe what we need is to change `< 0` to `<= 0` to avoid the rare case where fast spin gets decremented exactly to `0`?

Reply:
The fast spin only goes over the cap for a specific amount of time before it settles to the rotation speed cap. I've tested this in Golden Ratio for reference. If you don't want to count fast spin to go over the speed cap (as if you want to kill it completely), then feel free to make a suggestion and I'll accept it.

Reply:
OK, my misunderstanding was that I assumed the code above to be executed every frame. But I think that is more reasonable, so how about we keep the condition as:

```cpp
if(status.fastSpin <= 0 && abs(levelStatus.rotationSpeed) > rotationSpeedMax))
```

and move the entire `if` statement in the `updateRotation` function?